lib/power_action_utils.pm

```diff
@@ -81,12 +81,19 @@ sub reboot_x11 {
     my ($self) = @_;
     wait_still_screen;
     if (check_var('DESKTOP', 'gnome')) {
-        if (is_tumbleweed) {
+        # Get gnome version
+        x11_start_program('xterm -geometry 150x45+5+5', target_match => 'xterm');
+        become_root;
+        my $gnome_version = script_output("rpm -q --queryformat '%{VERSION}' gnome-shell");
+        type_string "killall xterm\n";
+
+        # Change in reboot process was introduced with gnome 3.38.0
+        if (package_version_cmp($gnome_version, '3.38.0') < 0) {
```

Comment:
I advise against doing something as low level as checking the package version. It should be enough to just turn around the logic check and do something like:

```perl
if (is_sle('<=15-sp2') || is_leap('<=15.2')) {
```

and not do the whole part about reading out RPM versions. No user would do that and the test should reflect clear user scenarios.

Reply:
Nonsense, You can't assure what gnome version will be on what version. What use would do has nothing to do with this issue that since gnome version 'x' reboot behavior changed.

Reply:
+1, in this case... gnome package doesn't change the behavior often enough between SLE/Leap versions, to need the change you're proposing, at least in the short future.

Reply:
I don't understand why you describe my comment as "nonsense". And regarding what @foursixnine wrote: What I am proposing is used in many other places, also to distinguish between different gnome versions. I don't see a need to do it more complicated here with more runtime code that is passed to the SUT when the test helper module is not about checking for specific package versions of gnome. Our tests should reflect clear user scenarios and be simple to follow and understand. We need to make something different, this is clear, but reading out an RPM version is IMHO a bad system level test design as we are messing internal details that a user does not need to care about. If you all still agree that this is the way to go, go ahead, I am not insisting.

Reply:
In general, I agree to follow what @okurz is proposing.

Usually, you have tests according to a specific version, so you don't need to check in the code which version are you using, but the test developer knows which version is to be tested and the test reflect that specific version in the test case. And if there are more versions available in the same product, you would have different test cases for each version.

There is no clean way to solve this kind of issue as long as the test code is trying to match at the same time every released and under development product.

Given what we have and in this case, I think it is worth to make an exception. So, I agree to check if the version is "smaller than".

Reply:
It should actually work to do the 'else' branch on all versions, without even caring for the version.

TW currently already uses this when it needs to reboot a Leap 42.x instance before it can reboot; In one case I needed to 'cheat' with a 'empty click which is not needed' - but essentially, it works.

And rebooting via the menu is imho more realistic to be used by users than ctrl-alt-del ever was.

Reply:
Because it's nonsense for Maintenance, for TW it's not a problem as it has latest package versions and there is not TW 12 15 ... with older gnome versions. It's problem for e.g. SLE 12-SP2, SLE 15 etc, some versions can get never gnome version tomorrow, some in one month or never, there is no connetion between openQA VERSION and gnome version.

Reply:
Any service pack of SLE will not suddenly get a new major version of gnome and even a minor version (not patch) should be done explicitly so even if the test fails then this is a good sign that the behaviour should be checked. If openQA tests are confused think about customers that would be confronted with a new gnome version pulled in by *maintenance patches*

Reply:
I'm curious why you think service pack won't get new version of gnome ?
This is not first time that interface or behavior of program or DM changed, I'm surprised we still got customers after so many visual changes between versions. You want to check expected behavior, with this view we should drop automation.

Reply:
I mean no new major version of gnome as maintenance patches during the life time of a service pack

Reply:
I don't know if there is any relation between service pack and gnome major version, and if it has to be major version patch for change like this one, I guess no.
With version of distribution there will be failing tests until the update is released. Also TW has staging where will be different gnome version. For maintenance with older supported SLE versions, while each version can get gnome update at any time, will be check of distribution version problem and breaking review.